ui/components/multichain/network-list-menu/popular-network-list/popular-network-list.tsx

```diff
@@ -0,0 +1,189 @@
+              <a
+                key="link"
+                className="add-network__edge-case-box__link"
+                href="https://chainlist.wtf/"
+                target="_blank"
+                rel="noreferrer"
+              >
+                {t('here')}.
+              </a>,
+              <ButtonPrimary
+                textAlign={TextAlign.Center}
+                variant={TextVariant.bodyMd}
+                size={ButtonPrimarySize.Md}
+                width={BlockSize.Full}
+                onClick={(event: React.MouseEvent<HTMLButtonElement>) => {
+                  event.preventDefault();
+                  if (isPopUp) {
+                    // eslint-disable-next-line @typescript-eslint/ban-ts-comment, spaced-comment
+                    //@ts-ignore
+                    platform.openExtensionInBrowser(ADD_NETWORK_ROUTE);
+                  } else {
+                    history.push(ADD_NETWORK_ROUTE);
+                  }
+                }}
+              >
+                {t('addMoreNetworks')}
+              </ButtonPrimary>,
+            ])}
+          </Text>
+        </Box>
+      </Box>
+    );
+  }
+
+  return (
+    <Box className="new-network-list__networks-container">
+      <Box
+        marginTop={isPopUp ? 0 : 4}
+        marginBottom={1}
+        paddingLeft={4}
+        paddingRight={4}
+      >
+        <Box
+          marginTop={4}
+          marginBottom={8}
+          display={Display.Flex}
+          justifyContent={JustifyContent.spaceBetween}
+        >
+          <Text> {t('additionalNetworks')}</Text>
+        </Box>
+        {searchAddNetworkResults.map((item: RPCDefinition, index: number) => (
+          <Box
+            key={index}
+            display={Display.Flex}
+            alignItems={AlignItems.center}
+            justifyContent={JustifyContent.spaceBetween}
+            marginBottom={6}
+            className="new-network-list__list-of-networks"
+          >
+            <Box display={Display.Flex} alignItems={AlignItems.center}>
+              <AvatarNetwork
+                size={AvatarNetworkSize.Md}
+                src={item.rpcPrefs?.imageUrl}
+                name={item.nickname}
+              />
+              <Box marginLeft={2}>
+                <Text
+                  color={TextColor.textDefault}
+                  backgroundColor={BackgroundColor.transparent}
+                  ellipsis
+                >
+                  {item.nickname}
+                </Text>
+              </Box>
+            </Box>
+            <Box
+              display={Display.Flex}
+              alignItems={AlignItems.center}
+              marginLeft={1}
+            >
+              <Button
+                type={ButtonVariant.Link}
+                className="add-network__add-button"
+                variant={ButtonVariant.Link}
+                data-testid="test-add-button"
+                onClick={async () => {
+                  dispatch(toggleNetworkMenu());
```

Comment:
Let's do this call second so that it disappears in the background instead of before the `requestUserApproval` finishes.

Reply:
eeem i think we will have a problem if we do it that way, The issue arises because `requestUserApproval` is an asynchronous function. When we use `await requestUserApproval()`, the execution pauses, waiting for the user to either approve or reject through the confirmation modal. This means the promise stays in a pending state until the user interacts with the "add network" confirmation modal, which is triggered by `requestUserApproval`. However, there's a complication: our modal ends up obscuring the confirmation modal because `dispatch(toggleNetworkMenu());` is called after, not before, this interaction. As a result, from a functional standpoint, we encounter a problem where the user might not be able to see or interact with the confirmation modal as intended.